_LowLevelCode/cpp/test/combine_sqw.tests/combine_sqw.test.cpp

```diff
@@ -0,0 +1,162 @@
+#include "combine_sqw/combine_sqw.h"
+#include "combine_sqw/nsqw_pix_reader.h"
+#include "combine_sqw/pix_mem_map.h"
+#include "combine_sqw/sqw_pix_writer.h"
+#include "test/utility/environment_helpers.h"
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+class pix_map_tester : public pix_mem_map {
+public:
+  void read_bins(std::size_t num_bin,
+                 std::vector<pix_mem_map::bin_info> &buffer,
+                 std::size_t &bin_end, std::size_t &buf_end) {
+    pix_mem_map::_read_bins(num_bin, buffer, bin_end, buf_end);
+  }
+  void read_bins_job() { pix_mem_map::read_bins_job(); }
+
+  std::size_t get_first_thread_bin() const { return n_first_rbuf_bin; }
+  std::size_t get_last_thread_bin() const { return rbuf_nbin_end; }
+  std::size_t get_n_buf_pix() const { return rbuf_end; }
+  void wait_for_read_completed() {
+    std::unique_lock<std::mutex> data_ready(this->exchange_lock);
+    this->bins_ready.wait(data_ready, [this]() { return this->nbins_read; });
+  }
+  bool thread_get_data(std::size_t &num_bin, std::vector<bin_info> &inbuf,
+                       std::size_t &bin_end, std::size_t &buf_end) {
+    return pix_mem_map::_thread_get_data(num_bin, inbuf, bin_end, buf_end);
+  }
+  void thread_query_data(std::size_t &num_first_bin, std::size_t &num_last_bin,
+                         std::size_t &buf_end) {
+    pix_mem_map::_thread_query_data(num_first_bin, num_last_bin, buf_end);
+  }
+  void thread_request_to_read(std::size_t start_bin) {
+    pix_mem_map::_thread_request_to_read(start_bin);
+  }
+};
+
+class TestCombineSQW : public ::testing::Test {
+
+protected:
+  static std::vector<uint64_t> sample_npix;
+  static std::vector<uint64_t> sample_pix_pos;
+  static std::vector<float> pixels;
+  static std::string test_file_name;
+  static std::size_t num_bin_in_file, bin_pos_in_file, pix_pos_in_file;
+
+  // Called once, before the first test is executed.
+  static void SetUpTestSuite() {
+    const auto HORACE_ROOT_DIR = get_env_variable("HORACE_ROOT");
```

Comment:
would it be helpful to define `HORACE_ROOT` as a const somewhere? that somewhere would then be a good reference for "these are the environment variables we're going to use in the code"